pkg/utils/predicates.go

```diff
@@ -322,11 +322,21 @@ func CheckPodsWithAntiAffinityExist(pod *v1.Pod, pods map[string][]*v1.Pod, node
 			}
 			for namespace := range namespaces {
 				for _, existingPod := range pods[namespace] {
-					if existingPod.Name != pod.Name && podMatchesTermsNamespaceAndSelector(existingPod, namespaces, selector) {
-						node, ok := nodeMap[pod.Spec.NodeName]
-						if !ok {
-							continue
+					if existingPod.Name == pod.Name || !podMatchesTermsNamespaceAndSelector(existingPod, namespaces, selector) {
+						continue
+					}
+
+					node, ok := nodeMap[pod.Spec.NodeName]
+					if !ok {
+						continue
+					}
+
+					if nodeFit {
+						if _, ok := node.Labels[term.TopologyKey]; ok {
```

Comment:
Here we still do not know whether `pod` and `existingPod` are violating the anti-affinity rule. Just that both pods are in anti-affinity relation. Checking whether `pod` is in a given topology domain is not sufficient here.

Reply:
In that case `nodeHavingExistingPod.Label` instead of `node.Labels`.